### ISSUE TITLE
Implement QASM interpreter.

### ISSUE THREAD

**Issue by vtomole** [vtomole/qc-hack-isu#1]
Make it easier to benchmark different interpreters.

**Comment by AlyShmahell** [comment]
It would be easier to help if you could share the resources used to learn the stuff about quantum mechanics that are present in the emulator. There are many great sources out there on wave functions and measurement, but I want the option where it is done in 24 hours :smile:

**Comment by vtomole** [comment]
This can't be done in 24 hours; but if you want to work on it, `vm.py` uses https://courses.cs.washington.edu/courses/csep590/05su/ as a reference.

**Comment by vtomole** [comment]
Here is the QASM spec: https://arxiv.org/abs/1707.03429

**Comment by vtomole** [comment]
@AlyShmahell I will guide you with any questions you have :)

**Comment by AlyShmahell** [comment]
@vtomole Thank you so much, I mentioned 24 hours because of your blog post :smile: 
The architecture seems understandable, I think I know the correspondence between the code and the spec, I will give it a deeper look in the next few days and ask all my questions in a batch.
The course is amazing, I think I will use the exercises as a guide to what to read about, did you take that course?!

**Comment by vtomole** [comment]
No, I just read the notes. I haven't taken an official quantum computing course.
